src/main/java/frc/robot/RobotContainer.java

```diff
@@ -50,41 +49,119 @@ public RobotContainer() {
         configureBindings();
     }
 
-    private void configureBindings() {
-        drive.registerTelemetry(logger::telemeterize);
-        drive.setDefaultCommand(new DriveWithJoysticks(drive, leftJoystick, rightJoystick));
+    private void configureSubsystems() {
+        // TODO: Potentially migrate to Constants.mode
+        if (Robot.isReal()) {
+            if (FeatureFlags.runDrive) {
+                drive =
+                        new PhoenixDrive(
+                                PhoenixDriveConstants.DrivetrainConstants,
+                                PhoenixDriveConstants.FrontLeft,
+                                PhoenixDriveConstants.FrontRight,
+                                PhoenixDriveConstants.BackLeft,
+                                PhoenixDriveConstants.BackRight);
+                logger = new Telemetry(6);
+            }
+            if (FeatureFlags.runVision) {
+                tagVision = new VisionLocalizer(new CameraContainerReal(VisionConstants.cameras));
+            }
+            if (FeatureFlags.runIntake) {
+                intakeSubsystem = new IntakeSubsystem(new IntakeNEOVortex());
+            }
+            if (FeatureFlags.runScoring) {
+                scoringSubsystem =
+                        new ScoringSubsystem(new ShooterIOTalonFX(), new AimerIORoboRio());
+            }
+        } else {
+            if (FeatureFlags.simulateDrive) {
+                drive =
+                        new PhoenixDrive(
+                                PhoenixDriveConstants.DrivetrainConstants,
+                                PhoenixDriveConstants.FrontLeft,
+                                PhoenixDriveConstants.FrontRight,
+                                PhoenixDriveConstants.BackLeft,
+                                PhoenixDriveConstants.BackRight);
+
+                logger = new Telemetry(6);
+            }
+            if (FeatureFlags.simulateVision) {
+                if (logger != null) {
```

Comment:
without the else, why not just add the vision initialization inside of the simulate drive conditional?

Reply:
I'd prefer to keep it structured consistently just because it's easy on the eyes and it'll be easier to find stuff if things are in consistent places.